SampleScripts/PowerManagement-WorkloadDomain.ps1

```diff
@@ -671,16 +708,47 @@ Try {
                             Exit
                         }
                     }
-                    #VSAN shutdown wizard automation
-                    Set-VsanClusterPowerStatus -server $vcServer.fqdn -user $vcUser -pass $vcPass -cluster $cluster.name -PowerStatus clusterPoweredOff
-                    $esxiDetails = $esxiWorkloadCluster[$cluster.name]
-                    foreach ($esxiNode in $esxiDetails) {
-                        if ((Test-NetConnection -ComputerName $esxiNode.fqdn -Port 443).TcpTestSucceeded) {
-                            Write-PowerManagementLogMessage -Type ERROR -Message "$($esxiNode.fqdn) is still up. Check the FQDN or IP address, or the power state of the '$($esxiNode.fqdn)'." -Colour Red
-                            Exit
+					$esxiDetails = $esxiWorkloadCluster[$cluster.name]
+					
+                    # vSAN or VxRail Manager shutdown wizard automation.
+                    if ($vxRailDetails -ne "") {
+                        Write-PowerManagementLogMessage -Type INFO -Message "Invoke VxRail cluster shutdown $($vxRailDetails.fqdn) $vcUser, and $vcPass"
+                        Invoke-VxrailClusterShutdown -server $vxRailDetails.fqdn -user $vcUser -pass $vcPass
+                        Write-PowerManagementLogMessage -Type INFO -Message "Sleeping for 60 seconds before polling for ESXI hosts shutdown status check..."
```

Comment:
Wouldn't it be better to define the sleep time and then pass that to the message and Start-Sleep?

Reply:
addressed in commit a0e6a231b99ac37596cd1b07cf674e22d71bda11